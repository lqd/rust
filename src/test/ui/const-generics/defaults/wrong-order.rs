// revisions: full min
#![cfg_attr(full, feature(const_generics))] //[full]~WARN the feature `const_generics` is incomplete

struct A<T = u32, const N: usize> {
    //[min]~^ ERROR using type defaults and const parameters in the same parameter list
    //~^^ ERROR const parameters must be declared prior to defaulted parameters
    arg: T,
}

fn main() {}
