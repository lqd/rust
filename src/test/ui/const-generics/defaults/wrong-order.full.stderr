error: const parameters must be declared prior to defaulted parameters
  --> $DIR/wrong-order.rs:4:25
   |
LL | struct A<T = u32, const N: usize> {
   |         ----------------^-------- help: reorder the parameters: `<const N: usize, T = u32>`
   |
   = note: the expected order for parameters is: lifetimes, then consts and types, then defaulted parameters

warning: the feature `const_generics` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/wrong-order.rs:2:27
   |
LL | #![cfg_attr(full, feature(const_generics))]
   |                           ^^^^^^^^^^^^^^
   |
   = note: `#[warn(incomplete_features)]` on by default
   = note: see issue #44580 <https://github.com/rust-lang/rust/issues/44580> for more information

error: aborting due to previous error; 1 warning emitted

